### ISSUE TITLE
Spawning Character tags as a AI with a spawn point hard crashes Sapien when the character tag references a creature tag

### ISSUE THREAD

**Issue by stryking** [Joint-Issue-Tracker/Joint-Issue-Tracker#65]
### Issue type: Is this a bug or feature request?

Bug

### Which game's editing kit is affected by the issue?

HREK

### Which tools are affected by the issue?

- [ ] Guerilla (`guerilla.exe`)
- [ ] Foundation (`Foundation.exe`)
- [X] Sapien (`sapien.exe`)
- [X] Standalone (`halox_tag_test.exe`)
- [ ] Tool (`tool.exe`)
- [ ] Tool Fast (`tool_fast.exe` - unsupported)
- [ ] MegaloEdit (`MegaloEdit.exe`)
- [X] In-game (MCC)

### Issue description

Unable to use creatures for any AI functionality at all, hard crashes the engine. 

https://www.youtube.com/watch?v=OarwGjd7xgM

![image](https://user-images.githubusercontent.com/15280727/201799081-a9eb2498-4c9a-4907-929e-81838f6d9690.png)

### Reproduction steps

[civilian_male.zip](https://github.com/Joint-Issue-Tracker/Joint-Issue-Tracker/files/10007973/civilian_male.zip)

(Is missing the _civilian_male.animation_graph_ file, Make sure you add it from default HREK into the root folder of this tag, was too big for github)

Both tags are found in "_tags\objects\characters\civilian_male\ai_" and they're: 
"**civilian_male_biped.character**" for the biped AI 
"**civilian_male2.character**" for the creature AI

1. Open the default _levels\test\box\box.scenerio_ in Sapien
2. Edit types of tag
3. Add new character tags
4. Add the two character tags in _tags\objects\characters\civilian_male\ai_ **civilian_male_biped.character** & **civilian_male2.character**
5. In the hierarchy view, click on _Scenerio>AI>squads>"squads_0"_
6. Check the box, **initially placed** in the properties panel
7. Then continue down the hierarchy _Scenerio>AI>squads>"squads_0">Spawn Points>**NONE: spawn_points_0**_
8. Change the character type from **civilian_male** to **civilian_male_biped**
9. Press **ALT R** to reset the map and it should spawn the civilian biped character
10. Now in **NONE: spawn_points_0** change the character type to **civilian_male2** which is the creature character
11. When you press ALT R Sapien should crash

### Debug logs

[debug.txt](https://github.com/Joint-Issue-Tracker/Joint-Issue-Tracker/files/10007879/debug.txt)
[debug_full.txt](https://github.com/Joint-Issue-Tracker/Joint-Issue-Tracker/files/10007880/debug_full.txt)
[crash_info.txt](https://github.com/Joint-Issue-Tracker/Joint-Issue-Tracker/files/10007882/crash_info.txt)

### How often does the issue occur?

100%

### Issue impact

Unable to use creatures for any AI functionality at all, have to use biped which is much more expensive on the engine and unoptimal. 

### Additional comments

Would like to option to use creature functionality for Reborn as having 50+ biped in one scene is very taxing and limiting and we would benefit greatly from being able to use the cheaper creature functionality for AI. 

The potential to eventually use swarm functionality as well would be very useful. 

Thanks. 
Ryan.

**Comment by kornman00** [comment]
This isn't a bug -

`Could not locate biped for character objects\characters\civilian_male\ai\civilian_male2`

Actors in squad must be implemented using units (bipeds, although a vehicle may technically 'work'), they cannot be implemented using creatures. You would need to place creatures in the scenario as you would place other types of objects.

**Comment by Krevil** [comment]
Changed ticket to feedback due to this not being a bug.

**Comment by earthcrafterman** [comment]
Using creatures in squads works in H3. Was that a bug or did Bungie actively remove that functionality for Reach to simplify the codebase?

**Comment by stryking** [comment]
It's a feature in Halo 3, the infection forms use them.

**Comment by kornman00** [comment]
> Using creatures in squads works in H3. Was that a bug or did Bungie actively remove that functionality for Reach to simplify the codebase?

Ah, so the function (attaching resources to a specific zone) this is asserting in wasn't written exactly the same in H3, it was much simpler. For one reason or another in Reach this was changed so that it enumerated all the characters in their hierarchy (since chars can have parent chars) and their units to then add their model and used variant names (maybe this was to automate what designers would normally have to manually do with designer zones).

In the `actor_place()` code, they normally check if the character is a swarm, in which case a totally different code path is taken that uses creatures. But since Reach didn't have infection forms, I can see how this was possibly an oversight in handling (or just was never a concern). I think this specific code path could be fixed up to handle swarm characters and attaching their models/variants to ensure their resources are available in zones they appear in.

FWIW, when it comes to "is this a bug or not?", I generally look at whether it was supported functionality in the _game_ that shipped on the underlying engine. Since Flood weren't in Reach and no other squads were implemented with swarm characters, I wouldn't view this as a bug per-say, but a feature request to resurrect what was previously a supported workflow.

TL;DR I think this assert could be addressed and swarm characters (that use creatures) accounted for in a future update. There may be other creature fall out that may be less trivial to address though (just want to set some expectations this may not be a silver bullet)

Thanks for taking the time to write all this up and noting this _was_ working in H3 :).

**Comment by stryking** [comment]
Thanks Korn and hopefully this is something that can be looked into in the future, appreciate all the work your doing 👍.
